### PR TITLE
fix: properly continue playback when repeating current song

### PR DESCRIPTION
**Pull request by DaPigGuy**

### CODE REVIEW

File: src/player/index.ts [hunk 1]
```diff
@@ -318,10 +318,13 @@ export class MuzikaPlayer extends MuzikaMediaStream {
...
       const track = this.queue.repeat_or_next();
+      const is_repeating = this.queue.repeat === RepeatMode.ONE;
 
-      if (!track) {
-        // this means there is no track to play. Just reprepare the current track
+      if (!track || is_repeating) {
+        // Reprepare the current track if there is no next track, or if the current track will repeat
         this.stream_prepared(...previous_metadata);
+
+        if (is_repeating) this.resume();
```
Comment (DaPigGuy) [inline src/player/index.ts:327]: Currently with song repeat, `Queue.repeat_or_next` calls `Queue.change_position` which does nothing because of the `position != this._position || force` condition. If we pass `force=true`, that does give us continued playback but the player scale/timestamp labels are stuck at 0:00 as their values are set as `initial_seek_to ?? timestamp`. This is because during the Queue's `play` signal, `initial_seek_to` is set to `0`. When going to previous/next songs, the Stream's `uri_loaded_cb` would then call `resume`, but since no new URI is loading on repeat, this doesn't happen. 

TLDR: just calling `resume` here is simpler :sweat_smile:
